ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/AlignmentUtil.java

```diff
@@ -0,0 +1,357 @@
+		m.setPosition(de, position);
+		m.setDockArea(de, dockArea);
+	}
+
+	// Shift eligible children
+	private void shiftChildren(final DiagramModification m, final DiagramElement parentDe, final double childOffset) {
+		for (final Queryable q : parentDe.getChildren()) {
+			if (q instanceof DiagramElement && axisAlignmentUtil.isShiftable((DiagramElement) q)) {
+				final DiagramElement childDe = (DiagramElement) q;
+				setPositionAndDockArea(m, childDe, axisAlignmentUtil.getShiftPostion(childDe, childOffset));
+			}
+		}
+	}
+
+	// Any selected element must not be a descendant of any other selected element.
+	// Must be docked to appropriate area
+	static boolean getEnabled(final Function<DockArea, Boolean> isValidDockArea) {
+		final List<DiagramElement> selectedElements = AgeHandlerUtil.getSelectedDiagramElements();
+
+		// More than one diagram elements must be selected
+		if (selectedElements.size() < 2) {
+			return false;
+		}
+
+		for (final DiagramElement de : selectedElements) {
+			if (!isValidDockArea.apply(de.getDockArea()) || isAncestorSelected(de, selectedElements)) {
+				return false;
+			}
+		}
+
+		final List<DiagramElement> dockedElements = selectedElements.stream().filter(de -> de.getDockArea() != null)
+				.collect(Collectors.toList());
+		if (dockedElements.size() > 1) {
+			return validDockedElement(dockedElements);
+		}
+
+		return true;
+	}
+
+	// Docked elements must be docked to opposite sides if they have the same parent
+	private static boolean validDockedElement(final List<DiagramElement> dockedElements) {
+		for (int i = 0; i < (dockedElements.size() - 1); i++) {
+			final DiagramElement de = dockedElements.get(i);
+			if (hasSameParentAndDockArea(dockedElements, de, i)) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	private static boolean hasSameParentAndDockArea(final List<DiagramElement> dockedElements, final DiagramElement de,
+			int i) {
+		while (i < (dockedElements.size() - 1)) {
+			final DiagramElement tmp = dockedElements.get(++i);
+			if (de.getDockArea() == tmp.getDockArea() && de.getParent() == tmp.getParent()) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	static boolean isAncestorSelected(final DiagramElement de, final List<DiagramElement> selectedElements) {
+		Queryable parent = de.getParent();
+
+		while (parent != null) {
+			if (selectedElements.contains(parent)) {
+				return true;
+			}
+			parent = parent.getParent();
+		}
+
+		return false;
+	}
+
+	// Element that other elements will align with
+	public static AlignmentElement getPrimaryAlignmentElement(final List<AlignmentElement> elements) {
+		if (elements.size() == 0) {
+			return null;
+		}
+
+		return elements.get(elements.size() - 1);
+	}
+
+	static class AlignmentElement {
+		private final DiagramElement de;
+		private double relativeLocation; // Element's location relative to the diagram
+
+		public AlignmentElement(final DiagramElement de, final Function<DiagramElement, Double> getAxisValue) {
```

Comment:
Rename setDiagramRelativeLocation to caluclateAbsoluteLocation. Delegate to other constructor:

this(de, calculateAbsolutePosition(de, getAxisValue));